test/backend/datastores.impala.spec.js

```diff
@@ -0,0 +1,51 @@
+import {assert} from 'chai';
+
+import {DIALECTS} from '../../app/constants/constants.js';
+
+import {
+    query, connect, tables
+} from '../../backend/persistent/datastores/Datastores.js';
+
+
+const connection = {
+    dialect: DIALECTS.APACHE_IMPALA,
+    host: '35.184.155.127', // default: '127.0.0.1'
+    port: 21000,
+    database: 'plotly', // default: ''
+    timeout: 60
+};
+
+describe('Apache Spark:', function () {
+
+    it('connect succeeds', function() {
+        this.timeout(180 * 1000);
+        return connect(connection);
+    });
+
```

Comment:
could we add another test that bad credentials causes the connection to fail with an appropriate error message?

Reply:
and since this is sql-like, it would be good to add this to the set of routes tests here: https://github.com/plotly/falcon-sql-client/blob/4dbbde6bfcbd43c6753053d80a283c5338bbcba4/test/backend/routes.spec.js#L544-L995